closure/goog/dom/forms.js

```diff
@@ -318,21 +318,21 @@ goog.dom.forms.hasValueByName = function(form, name) {
  *     (or null).
  */
 goog.dom.forms.getValue = function(el) {
+  // Elements with a type may need more specialized logic.
   var type = /** @type {!HTMLInputElement} */ (el).type;
-  if (!goog.isDef(type)) {
-    return null;
-  }
-  switch (type.toLowerCase()) {
-    case goog.dom.InputType.CHECKBOX:
-    case goog.dom.InputType.RADIO:
-      return goog.dom.forms.getInputChecked_(el);
-    case goog.dom.InputType.SELECT_ONE:
-      return goog.dom.forms.getSelectSingle_(el);
-    case goog.dom.InputType.SELECT_MULTIPLE:
-      return goog.dom.forms.getSelectMultiple_(el);
-    default:
-      return goog.isDef(el.value) ? el.value : null;
+  if (goog.isDef(type)) {
```

Comment:
Rather than indenting this an extra step we can simply fold the check in directly:

  switch (type && type.toLowerCase()) {

This will also allow to continue using the default branch.